ivadomed/main.py

```diff
@@ -32,8 +32,20 @@ def get_parser():
 
     # MANDATORY ARGUMENTS
     mandatory_args = parser.add_argument_group('MANDATORY ARGUMENTS')
+    command_group = mandatory_args.add_mutually_exclusive_group(required=True)
+    
+    command_group.add_argument("--train", dest='train', action='store_true', 
+                                help="Perform training on data.")
+    command_group.add_argument("--test", dest='test', action='store_true', 
+                                help="Perform testing on trained model.")
+    command_group.add_argument("--segment", dest='segment', action='store_true', 
+                                help="Perform segmentation on data.")
     mandatory_args.add_argument("-c", "--config", required=True, type=str,
                                 help="Path to configuration file.")
+    mandatory_args.add_argument("-pd", "--path-data", required=True, type=str,
```

Comment:
Can we keep these arguments (path-data, path-output) optional and continue to support the version with all the parameters in the config file? Personally, I find it more convenient to have everything in a single config file, but I get why it can also be more convenient to externalize these flags. The best solution, in my opinion, would be to continue supporting all the options in the config file (for retrocompatibility and for dev convenience), but also add the option to overwrite some parameters in the CLI.

Reply:
Good points! I removed the entries from the config file because from my understanding the linked issue #615 implied to transition from config files to CLI flag. I can definitely change it to support both JSON files and and CLI flags. Wouldn't mind some other input from others as well before proceeding. Tagging @jcohenadad since he created the issue and also @ahill187 for visibility.

Reply:
Yah they're both good points, I see the advantage of supporting both methods, although having just one method is probably cleaner/simpler? Particularly the `train/test/segment` flags seem to make more sense as just command-line parameters? 
I suppose we could make the `path_bids` and `path_output` default to whatever directory you are currently in? I'm not sure if that helps at all though haha.

Reply:
One of my concerns would also be in terms of maintenance in the long-term. It may be confusing for future users if they can specify commands both via CLI and config files. Perhaps a better solution would be to support both in the short-term and consider deprecating the config option at a later date?

Reply:
Yah that's a good point about maintaining long-term, it's not ideal to have coupled code to maintain in tandem.

Reply:
If we have to choose only one of the methods, I think it would be better to keep everything in the configuration file since I find it confusing and inconvenient to do the back and forth between both. But, of course, this is really personal. I still think both options can coexist, just like what is done with `sct_run_batch` on SCT (https://spinalcordtoolbox.com/en/stable/user_section/command-line.html#sct-run-batch).

Reply:
Perhaps we could get some more input from other users in the lab? We could discuss on `Slack` or do a poll or something, or discuss at the next lab meeting?

Reply:
Yes, absolutely we can discuss this at the next lab meeting on Wednesday.